File: src/main/java/hzt/model/Resource.java
package hzt.model;

import org.jetbrains.annotations.NotNull;

import java.util.Objects;

public class Resource implements Comparable<Resource> {

    private final String name;
    private final String pathToResource;

    public Resource(String name) {
        this(name, "");
    }

    public Resource(String name, String pathToResource) {
        this.name = name;
        this.pathToResource = pathToResource;
    }

    public String getPathToResource() {
        return pathToResource;
    }

    @Override
    public int compareTo(@NotNull Resource o) {
        return name.compareTo(o.name);
    }

    @Override
    public boolean equals(Object o) {
        return this == o || (o instanceof Resource resource && Objects.equals(name, resource.name));
    }

    @Override
    public int hashCode() {
        return Objects.hash(name);
    }

    @Override
    public String toString() {
        return name;
    }

}

File: src/main/kotlin/hzt/model/Resource.kt
package hzt.model

import java.io.InputStream
import java.net.URL
import java.util.*

class Resource
@JvmOverloads constructor(private val name: String, val url: URL? = null) : Comparable<Resource> {

    override fun compareTo(other: Resource): Int {
        return name.compareTo(other.name)
    }

    override fun equals(other: Any?): Boolean {
        return this === other || other is Resource && name == other.name
    }

    @Throws(exceptionClasses = [IllegalStateException::class])
    fun <R> useInputStream(useInputStream: (InputStream) -> R) = url?.openStream()?.use {
        useInputStream(it)
    }

    override fun hashCode(): Int {
        return Objects.hash(name)
    }

    override fun toString(): String {
        return name
    }
}